sources/subs/Menu.class.php

```diff
@@ -766,4 +853,17 @@ public function callMenu($selectedMenu)
 			$selectedMenu['function']();
 		}
 	}
+
+	/**
+	 * Retrieve easily the sole instance of this class.
+	 *
+	 * @return Menu
+	 */
+	public static function instance()
```

Comment:
Yup ... best I can do!

I'm going to finish what I have with some test cases and then close this PR, I really don't think its worth changing all this ATM.